File: backend/dalathasfarm/src/main/java/com/example/dalathasfarm/dtos/UpdateUserDto.java
package com.example.dalathasfarm.dtos;

import jakarta.validation.constraints.NotBlank;
import lombok.*;

import java.time.LocalDate;
import java.util.Date;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class UpdateUserDto {
    @NotBlank(message = "Password không được bỏ trống")
    private String password;
    private String retypePassword;
    private String email;
    private String fullName;
    private String address;

    @NotBlank(message = "Số điện thoại không được bỏ trống")
    private String phoneNumber;

    private LocalDate dateOfBirth;
}

File: admin/src/app/dtos/user/update.user.dto.ts
export class UpdateUserDto {
    fullName: string;    
    address: string;    
    password: string;    
    retypePassword: string;    
    dateOfBirth: Date;    
    
    constructor(data: any) {
        this.fullName = data.fullName;
        this.address = data.address;
        this.password = data.password;
        this.retypePassword = data.retypePassword;
        this.dateOfBirth = data.dateOfBirth;        
    }
}